File: demo/src/main/java/com/example/demo/service/UserService.java
package com.example.demo.service;

import com.example.demo.entity.User;
import com.example.demo.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

@Transactional(readOnly = true)
@Service
public class UserService {

    private final UserRepository userRepository;


    @Autowired
    public UserService(UserRepository userRepository){
        this.userRepository = userRepository;
    }
    public List<User> getUsers(){

        return userRepository.findAll();
    } //문제 없음

    public List<User> getUserByName(String name){
        return userRepository.findByName(name);
    }

    public Optional<User> getUserByNickname(String nickname){ return userRepository.findByNickname(nickname);}

    @Transactional
    public void insertUser(User user){
        userRepository.save(user); //문제 없음
    }

    @Transactional
    public User modifyUser(User user) {
        return userRepository.save(user);
    }

    @Transactional
    public void deleteUser(Long id){
        userRepository.deleteById(id);
    }
}

File: testproj/src/main/kotlin/com/example/testproj/testproj/service/UserService.kt
package com.example.testproj.testproj.service

import com.example.testproj.testproj.entity.User
import com.example.testproj.testproj.repository.UserRepository
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service

@Service
class UserService(private val userRepository: UserRepository){
    fun getUsers() : List<User>{ //이게 맞는지 ResponseEntity가 맞는지?
        val users = userRepository.findAll()
        return users
    }

    fun setUser(user : User): String{
        try {
            userRepository.save(user)
        }catch(e:Exception){
            return e.toString()
        }
        return "200"
    }

    fun getUserNameByUId(uid: String) : String{
        val users = userRepository.findAll()
        for(user in users){
            if(user.uid.equals(uid)){
                return user.name
            }
        } //이건 안좋아
        return "not found exception"
    }
}